File: Estruturas de dados lineares/Pilha.java
class FILO {
    //dados
    int[] array;
    int n;

    //construtor
    public FILO() {
        this.array = new int[100];
        n = 0;
    }

    /**
     * 
     * @param tamanho
     */
    public FILO(int tamanho) {
        this.array = new int[tamanho];
        n = 0;
    }

    /**
     * 
     * @param elemento
     */
    public void empilhar(int elemento) throws Exception{
        if ( n == array.length )
        {
            throw new Exception("array cheio");
        }
        
        array[n] = elemento;
        n++;
    }

    /**
     * 
     * @return
     */
    public int desempilhar() throws Exception{
        if( n == 0 ) {
            throw new Exception("array vazio");
        }
        return array[--n];
    }

    public void mostrar() {
        System.out.print("[");
        for(int i = 0; i < n; i++ ) {
            System.out.print(" "+ array[i]);
        }

        System.out.println(" ]");
    }
}

public class Pilha {
    public static void main(String[] args) throws Exception {
        //dados
        final FILO pilha = new FILO(3);

        //exemplos de comando
        /*pilha.empilhar(1);
        pilha.empilhar(5);
        pilha.desempilhar();
        pilha.mostrar();*/
    }
}

File: Estruturas de dados lineares/C/Pilha.c
#include "stdio.h"
#include "stdlib.h"

#define MAXTAM 6

typedef struct {
    //dados
    int array[MAXTAM];
    int n;
} Pilha;

//funcões
void iniciar(Pilha *pilha);
void empilhar(int elemento, Pilha *pilha);
int desempilhar(Pilha *pilha);
void mostrar(Pilha *pilha);

int main(void) {
    //dados
    Pilha pilha;

    //exemplos de comando
    empilhar(1, &pilha);
    empilhar(5, &pilha);
    desempilhar(&pilha);
    mostrar(&pilha);
}

/*
    @param pilha
*/
void iniciar(Pilha *pilha) {
    pilha->n = 0;
}

/*
   @param elemento
   @param pilha
*/
void empilhar(int elemento, Pilha *pilha) {
    if ( pilha->n == MAXTAM ) {
        exit(1);
    }

    pilha->array[pilha->n++] = elemento;
}

/*
    @param pilha
*/
int desempilhar(Pilha *pilha) {
    if (pilha->n == 0 ) {
        exit(1);
    }

    pilha->n--;
    
    return pilha->array[pilha->n];
}

/*
    @param lista
*/
void mostrar(Pilha *pilha) {
    printf("%s", "[ ");

    for(int i = 0; i < pilha -> n; i++) {
        printf("%i %s", pilha -> array[i], "");
    }

    printf("%s", "]\n");
}